src/metabase/models/permissions.clj

```diff
@@ -320,10 +323,15 @@
    ;; based on value of read-or-write determine the approprite function used to calculate the perms path
    (let [path-fn (case read-or-write
                    :read  collection-read-path
-                   :write collection-readwrite-path)]
+                   :write collection-readwrite-path)
+         collection-id (if (and (contains? this :trashed_from_collection_id)
+                                (contains? this :archived)
```

Comment:
I am trying to figure out what this check does, but I think it's redundant with `(:archied this)`

1. {... :archived true}

The `(:archived this)` call will be truthy

2. {... :archived false}

The `(:archived this)` call will be falsey

3. {...} ;; no archived

The `(:archived this)` call will be falsey

---

Am I missing something?

Reply:
Good call! You're right, I'll take this out.